chain/network/src/routing/routing.rs

```diff
@@ -81,6 +82,23 @@ impl Edge {
         Edge(Arc::new(EdgeInner::new(peer0, peer1, nonce, signature0, signature1)))
     }
 
+    pub fn nonce(&self) -> u64 {
+        self.0.nonce
+    }
+
+    pub fn signature0(&self) -> &Signature {
+        &self.0.signature0
+    }
+
+    pub fn signature1(&self) -> &Signature {
+        &self.0.signature1
+    }
+
+    pub fn removal_info(&self) -> &Option<(bool, Signature)> {
+        &self.0.removal_info
+    }
```

Comment:
`&Option<T>` isn't an idiomatic type to return from the getter, `Option<&T>` is. 

```suggestion
    pub fn removal_info(&self) -> Option<&(bool, Signature)> {
        self.0.removal_info.as_ref()
    }
```